tunein/__init__.py

```diff
@@ -55,24 +60,41 @@ def get_stream_url(url):
                 else:
                     return url
 
+    @staticmethod
+    def featured():
+        res = requests.post(TuneIn.featured_url)
+        yield from TuneIn._get_stations(res)
+
     @staticmethod
     def search(query):
         res = requests.post(TuneIn.search_url, data={"query": query})
+        yield from TuneIn._get_stations(res)
+
+    @staticmethod
+    def _get_stations(res: requests.Response):
         res = xml2dict(res.text)
         if not res.get("opml"):
             return
-        for entry in res['opml']['body']["outline"]:
+        # stations might be nested based on Playlist/Search
+        nested = res['opml']['body']["outline"][0].pop("outline", [])
+        if nested:
+            stations = nested
+        else: 
+            stations = res['opml']['body'].pop("outline", [])  
+
+        for entry in stations:
             try:
-                if entry.get("type") == "audio" and entry.get("item") == "station":
+                if not entry.get("key") == "unavailable" \
+                        and entry.get("type") == "audio" \
+                        and entry.get("item") == "station":
                     yield TuneInStation(
                         {"stream": TuneIn.get_stream_url(entry["URL"]),
                          "url": entry["URL"],
                          "title": entry.get("current_track") or entry.get("text"),
                          "artist": entry.get("text"),
                          "description": entry.get("subtext"),
-                         "image": entry.get("image"),
-                         "query": query
```

Comment:
keep the query, while not used by OCP consuming this data it is part of the previous api, other projects using this package might be expecting it for some reason